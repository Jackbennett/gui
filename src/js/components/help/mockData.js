export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
