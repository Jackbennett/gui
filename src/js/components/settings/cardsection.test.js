import React from 'react';
import renderer from 'react-test-renderer';
import { loadStripe } from '@stripe/stripe-js';
import { Elements } from '@stripe/react-stripe-js';
import CardSection from './cardsection';
import { undefineds } from '../../../../tests/mockData';

describe('GlobalSettings Component', () => {
  it('renders correctly', () => {
    jest.mock('@stripe/stripe-js', () => ({
      loadStripe: () => ({ createPaymentMethod: jest.fn() })
    }));
    const stripe = loadStripe();
    const tree = renderer
      .create(
        <Elements stripe={stripe}>
          <CardSection />
        </Elements>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
