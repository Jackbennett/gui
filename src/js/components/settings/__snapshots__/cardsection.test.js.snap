// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalSettings Component renders correctly 1`] = `
<form
  class="margin-top-small"
>
  <div
    class="warning"
  />
  <div
    id="poweredByStripe"
  >
    <div>
      All standard credit card fees apply (e.g. foreign transaction fee, if your card has one)
    </div>
    <img
      src="powered_by_stripe.png"
    />
  </div>
  <p
    class="info"
  >
    Billing will be scheduled monthly, starting from today. You can cancel at any time.
  </p>
  <div
    class="flexbox center-aligned margin-top-small"
    style="justify-content: flex-end;"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary Mui-disabled Mui-disabled"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Sign up
      </span>
    </button>
  </div>
  <div
    class="loaderContainer shrunk"
  >
    <div
      class="  loader"
    >
      <span
        class="dot dot_1"
      />
      <span
        class="dot dot_2"
      />
      <span
        class="dot dot_3"
      />
      <span
        class="dot dot_4"
      />
    </div>
  </div>
</form>
`;
